delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Projects.scala

```diff
@@ -159,6 +165,44 @@ trait Projects {
     * @param offset the last seen event offset; it will not be emitted by the stream
     */
   def currentEvents(offset: Offset = NoOffset): Stream[Task, Envelope[ProjectEvent]]
+
+  /**
+    * Create a project for a user according to automatic provisioning settings.
+    *
+    * @param subject  the user for which to provision the project.
+    */
+  def provisionProject(subject: Subject): UIO[Unit]
```

Comment:
`Projects` is a building block that is used everywhere and adding this in project complicates tests and other uses of `Projects` where we don't care about provisionning a project.
It would be better to create a higher level component (We could call it `ProjectProvisioning`) like we do right now for resolving/indexing resources.
So no impact on the tests and the dummy and it would isolate this portion of code.